plugins/GSdx/GSState.cpp

```diff
@@ -465,6 +465,21 @@ GSVector4i GSState::GetFrameRect(int i)
 	return rectangle;
 }
 
+int GSState::GetFramebufferHeight()
+{
+	const GSVector4i output[2] = { GetFrameRect(0), GetFrameRect(1) };
+	// Framebuffer height is 11 bits max according to GS user manual
+	const int height_limit = (1 << 11) - 1;
```

Comment:
It is true that rendering is limited to a square of 2048x2048. However I'm not sure that reading back the memory is limited by the same limit. I guess the limit depends on the window offset and the maximum pixels that can be read in x/y direction.